.github/workflows/precommit.yml

```diff
@@ -0,0 +1,49 @@
+name: Formatting & compliance
+
+on:
+  push:
+    branches:
+    - actions
+  pull_request:
+    branches:
+    - actions
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9.7
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        miniconda-version: "latest"
+        channels: conda-forge
+        channel-priority: flexible
+        show-channel-urls: true
+    - name: Install dependencies
+      run: |
+        conda create -n blueprint
+        conda activate blueprint
+        conda install --yes --file \{\{cookiecutter.project_slug\}\}/requirements/requirements.in
+        conda deactivate
+    - name: Install dev dependencies
+      run: |
+        conda create -n dev_blueprint
+        conda activate dev_blueprint
+        conda install --yes --file \{\{cookiecutter.project_slug\}\}/requirements/dev-requirements.in
```

Comment:
Here, the runtime requirements in `requirements.in` should be installed before those in `dev-requirements.in`, shouldn't they? Otherwise, you only have the dev tools but are missing the actual runtime requirements...

Reply:
good catch, that was a missunderstanding of the conventions from my side. 
I understood that dev-requirements should contain all dependencies of requirements.
I changed it in both github actions and jenkins

Reply:
> I understood that dev-requirements should contain all dependencies of requirements.
No, that only applies to `dev-environment.yml`, which is a superset of `environment.yml`